trillian/integration/ct_integration.go

```diff
@@ -168,6 +169,69 @@ func (t *testInfo) checkInclusionOf(ctx context.Context, chain []ct.ASN1Cert, sc
 	return nil
 }
 
+// checkInclusionOfPreCert checks a pre-cert is included at given index.
+func (t *testInfo) checkInclusionOfPreCert(ctx context.Context, tbs []byte, issuer *x509.Certificate, sct *ct.SignedCertificateTimestamp, sth *ct.SignedTreeHead) error {
+	// Calculate leaf hash =  SHA256(0x00 | tls-encode(MerkleTreeLeaf))
+	leaf := ct.MerkleTreeLeaf{
+		Version:  ct.V1,
+		LeafType: ct.TimestampedEntryLeafType,
+		TimestampedEntry: &ct.TimestampedEntry{
+			Timestamp: sct.Timestamp,
+			EntryType: ct.PrecertLogEntryType,
+			PrecertEntry: &ct.PreCert{
+				IssuerKeyHash:  sha256.Sum256(issuer.RawSubjectPublicKeyInfo),
+				TBSCertificate: tbs,
+			},
+			Extensions: sct.Extensions,
+		},
+	}
+	leafData, err := tls.Marshal(leaf)
+	if err != nil {
+		return fmt.Errorf("tls.Marshal(precertLeaf)=(nil,%v); want (_,nil)", err)
+	}
+	hash := sha256.Sum256(append([]byte{merkletree.LeafPrefix}, leafData...))
+	rsp, err := t.client().GetProofByHash(ctx, hash[:], sth.TreeSize)
+	t.stats.done(ctfe.GetProofByHashName, 200)
+	if err != nil {
+		return fmt.Errorf("got GetProofByHash(sct, size=%d)=nil,%v", sth.TreeSize, err)
+	}
+	fmt.Printf("%s: Inclusion proof leaf %d @ %d -> root %d = %x\n", t.prefix, rsp.LeafIndex, sct.Timestamp, sth.TreeSize, rsp.AuditPath)
+	if err := Verifier.VerifyInclusionProof(rsp.LeafIndex, int64(sth.TreeSize), rsp.AuditPath, sth.SHA256RootHash[:], leafData); err != nil {
+		return fmt.Errorf("got VerifyInclusionProof(%d,%d,...)=%v; want nil", rsp.LeafIndex, sth.TreeSize, err)
+	}
+	if err := t.checkStats(); err != nil {
+		return fmt.Errorf("unexpected stats check: %v", err)
+	}
+	return nil
+}
+
+// checkPreCertEntry retrieves a pre-cert from a known index and checks it.
+func (t *testInfo) checkPreCertEntry(ctx context.Context, precertIndex int64, tbs []byte) error {
+	precertEntries, err := t.client().GetEntries(ctx, precertIndex, precertIndex)
+	t.stats.done(ctfe.GetEntriesName, 200)
+	if err != nil {
+		return fmt.Errorf("got GetEntries(%d,%d)=(nil,%v); want (_,nil)", precertIndex, precertIndex, err)
+	}
+	if len(precertEntries) != 1 {
+		return fmt.Errorf("len(entries)=%d; want %d", len(precertEntries), 1)
+	}
+	leaf := precertEntries[0].Leaf
+	ts := leaf.TimestampedEntry
+	fmt.Printf("%s: Entry[%d] = {Index:%d Leaf:{Version:%v TS:{EntryType:%v Timestamp:%v}}}\n",
+		t.prefix, precertIndex, precertEntries[0].Index, leaf.Version, ts.EntryType, timeFromMS(ts.Timestamp))
+
+	if ts.EntryType != ct.PrecertLogEntryType {
```

Comment:
Sorry, too much brevity!
I was suggesting:
```golang
if got, want := ts.EntryType, ct.PrecertLogEntryType; got != want {
  return fmt.Errorf("...; want %v", precertIndex, got, want)
}
```
to avoid the potential for future drift between what's being tested, and what's being reported.